src/jtag3.c

```diff
@@ -1421,6 +1421,11 @@ static int jtag3_initialize(const PROGRAMMER *pgm, const AVRPART *p) {
       /* Partial Family_ID has been returned */
       pmsg_notice("partial Family_ID returned: \"%c%c%c%c\"\n",
                   resp[3], resp[4], resp[5], resp[6]);
+
+      // Read chip silicon revision
+      char chip_rev[AVR_CHIP_REVLEN];
```

Comment:
Should this better be guarded like so
```
      if(pgm->read_chip_rev) {
        char chip_rev[AVR_CHIP_REVLEN];
        pgm->read_chip_rev(pgm, p, chip_rev);
        pmsg_notice("silicon revision: %x.%x\n", chip_rev[0] >> 4, chip_rev[0] & 0x0f);
      }
```

Reply:
It is guarded by `if (p->prog_modes & PM_UPDI)`, but I can move the function call outside this and do it your way instead. It appears that Xmegas also have a revision number that can be read using a programmer. However, this is done a little differently, as the REVID byte is the next byte in memory after the signature. Perhaps we should add `revid` as a memory we can read from, and add it to the `read_chip_rev()` function?

From the Xmega A manual:

<img width="784" alt="image" src="https://github.com/avrdudes/avrdude/assets/3238730/a5bb92ab-2439-4d52-aa47-971fff4f7eb2">

<img width="779" alt="image" src="https://github.com/avrdudes/avrdude/assets/3238730/7a13c95e-1566-4a15-9145-f5ad0ecc6e10">